Refactor for scraped

ondenman opened the pull request:
## [Scraper Change checklist](https://github.com/everypolitician/everypolitician/wiki/Scraper-Change-checklist)
* [x] 1. scraper is on Morph.io under the "everypolitician-scrapers" group?
* [x] 2. scraper's GitHub "Website" link points at morph.io page?
* [x] 3. scraper is set to auto-run?
* [ ] 4. ~scraper is archiving?~ Not yet. This step will be done in a forthcoming PR.
* [x] 5. legislature has a scraper webhook set?

## [Gemfile checklist](https://github.com/everypolitician/everypolitician/wiki/Gemfile-checklist/)
* [x] 1. all links are secure
* [x] 2. links to Github use `github:` protocol, not simply `git:`
* [x] 3. formatting is consistent with our normal Rubocop setup

ondenman committed:
Use require_all

ondenman committed:
Bundle update scraped

ondenman committed:
Extract MemberPage

ondenman committed:
Extract MemberRow

ondenman committed:
Extract MembersPage

ondenman committed:
Merge MemberPage and MemberRow hashes

ondenman committed:
Remove superfluos methods

tmtmtmtm reviewed (changes requested):
I haven't examined the code yet, but if this is simply a refactoring, then I would expect the results to be the same before and after, which they're not.

You've renamed 'party' to 'party_name', and introduced a new 'member_url' field.

(NB: the 'source' field looks like it _should_ contain what you have in 'member_url' — i.e. the page about the person themselves, rather than the list page. But if that's currently broken, then you shouldn't fix that in a refactoring. Instead leave a comment, with the URL of the github issue, and then we can go back and fix that in a separate PR)

ondenman committed:
fixup! Extract MemberRow

ondenman committed:
fixup! Extract MemberRow

ondenman committed:
Rename member_url field to source

Stores each member's page URL.

ondenman committed:
Use absolute URLs decorator

tmtmtmtm commented on scraper.rb:
```diff
@@ -6,18 +6,12 @@
 require 'nokogiri'
 require 'open-uri'
```
You've left behind quite a few unnecessary 'require's here — e.g. you removed the function that uses 'date', but not removed the `require date`

tmtmtmtm commented on scraper.rb:
```diff
@@ -32,29 +26,14 @@ def datefrom(date)
+  MembersPage.new(response: Scraped::Request.new(url: url).response)
+             .member_rows
+             .each do |row|
+    mp_page = MemberPage.new(response: Scraped::Request.new(url: row.source).response)
```
The staggered indentation makes this quite difficult to read. I'd suggest using the `scrape` helper method that we use in other places, so this can be simplified:

```ruby
def scrape(h)
  url, klass = h.to_a.first
  klass.new(response: Scraped::Request.new(url: url).response)
end
```

ondenman committed:
Remove superfluous requires

We are no longer using the date gem and nokogiri and open-uri come
with scraped.

ondenman committed:
Move scraped object creation into method

Replaces repetitive code and makes it easier to follow.

ondenman committed:
Clear out databse before scraping

tmtmtmtm commented:
Woot!
